Simplified/NSDate+NYPLDateAdditions.m

```diff
@@ -20,6 +20,25 @@ + (NSDate *)dateWithRFC3339String:(NSString *const)string
   return date;
 }
 
++ (NSDate *)dateWithISO8601DateString:(NSString *const)string
+{
+  NSISO8601DateFormatter *const ISODateFormatter = [[NSISO8601DateFormatter alloc] init];
+  
+  ISODateFormatter.formatOptions = NSISO8601DateFormatWithFullDate;
+  ISODateFormatter.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
+
+  NSDate *const date = [ISODateFormatter dateFromString:string];
+
+  // in case date came in as a year only format
```

Comment:
So we try and make a date from the ISO formatter. If that ends up being nil it means we've failed. So we try again by guessting that we have a year format like "yyyy". That could also fail. Which is fine but let's make sure that a `nil` propogating down is tested. (The comment is misleading. I'd just remove it).

Reply:
Easy enough. I've removed the comment!